server/controller/authentication.go

```diff
@@ -234,6 +243,60 @@ func (c *Controller) loginEndpoint(ctx echo.Context) error {
 	}
 }
 
+func (c *Controller) postMultifactor(ctx echo.Context) error {
+	var request struct {
+		TOTP string `json:"totp"`
+	}
+	if err := ctx.Bind(&request); err != nil {
+		return c.wrapAndReturnError(ctx, err, http.StatusBadRequest, "malformed json")
+	}
+
+	repo := c.mustGetAuthenticatedRepository(ctx)
+	me, err := repo.GetMe(c.getContext(ctx))
+	if err != nil {
+		c.updateAuthenticationCookie(ctx, ClearAuthentication)
+		return c.wrapAndReturnError(ctx, err, http.StatusInternalServerError, "unable to retrieve current user")
+	}
+
+	if err := me.Login.VerifyTOTP(request.TOTP, c.clock.Now()); err != nil {
+		return c.returnError(ctx, http.StatusUnauthorized, "invalid TOTP code")
+	}
```

Comment:
Double check to see if this will remove the cookie, we need to keep that around. We dont want to remove it if they get it wrong, it should be good for a few minutes.